src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1635,6 +1639,40 @@ else if (null == currentPrimaryPlaceHolder) {
                     currentConnectPlaceHolder = currentPrimaryPlaceHolder;
                 }
 
+                try {
+                    InetAddress[] inetAddrs = InetAddress.getAllByName(currentConnectPlaceHolder.getServerName());
+
+                    // if only one IP is resolved and user didn't set TNIR explicitly then set TNIR to false
+                    if ((1 == inetAddrs.length) && !userSetTNIR) {
```

Comment:
We still want to keep the TIMEOUTSTEP and TIMEOUTSTEP_TNIR where we have different timeslices for TNIR and also turn TNIR off for FedAuth. 
If you think that it is cleaner to have a new PR, then I am fine with it. 

Basically the solution right now is 
1. Turn Tnir off for Fed Auth
2. Change TNIR time slices
3. Keep TNIR on if the user has explicitly specified it in the connection string.